internal/action/actions.go

```diff
@@ -412,6 +412,48 @@ func (h *BufPane) ParagraphNext() bool {
 	return true
 }
 
+// SelectParagraphPrevious select to the previous empty line, or beginning of the buffer if there's none
+func (h *BufPane) SelectParagraphPrevious() bool {
+	var line int
+	if !h.Cursor.HasSelection() {
+		h.Cursor.OrigSelection[0] = h.Cursor.Loc
+	}
+	for line = h.Cursor.Y; line > 0; line-- {
+		if len(h.Buf.LineBytes(line)) == 0 && line != h.Cursor.Y {
+			h.Cursor.X = 0
+			h.Cursor.Y = line
+			break
+		}
+	}
+	// If no empty line found. move cursor to end of buffer
+	if line == 0 {
+		h.Cursor.Loc = h.Buf.Start()
+	}
+	h.Cursor.SelectTo(h.Cursor.Loc)
```

Comment:
`h.Relocate()` is missing. As a result, when selecting paragraphs beyond the bottom/top of the screen, it doesn't scroll the screen automatically (unlike with `ParagraphNext` and `ParagraphPrevious`).

...In fact, looks like this can be implemented much shorter, by reusing `ParagraphNext()` and `ParagraphPrevious()` (and that would also automatically take care of scrolling):

```
// SelectParagraphPrevious select to the previous empty line, or beginning of the buffer if there's none
func (h *BufPane) SelectParagraphPrevious() bool {
	if !h.Cursor.HasSelection() {
		h.Cursor.OrigSelection[0] = h.Cursor.Loc
	}
	h.ParagraphPrevious()
	h.Cursor.SelectTo(h.Cursor.Loc)
	return true
}

// SelectParagraphNext select to the next empty line, or end of the buffer if there's none
func (h *BufPane) SelectParagraphNext() bool {
	if !h.Cursor.HasSelection() {
		h.Cursor.OrigSelection[0] = h.Cursor.Loc
	}
	h.ParagraphNext()
	h.Cursor.SelectTo(h.Cursor.Loc)
	return true
}
```

...Actually there is one more issue: if there are several empty lines one after another, it just selects them one by one, as if they were "paragraphs", instead of jumping to selecting the next paragraph. But the issue actually exists with `ParagraphNext` and `ParagraphPrevious` as well, so it can be fixed separately.

Reply:
In this solution, while it does select the block, the cursor does not get moved with the selection, rather it is projected from the cursor while it stays in place. I would expect block select to drag the selection behind the cursor, essentially being `ParagraphNext` + selection. In the same way that ctrl+shift+left/right moves the cursor, and all other ctrl+shift combinations.